xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/bridge/WikiComponentBridge.java

```diff
@@ -120,4 +122,13 @@
      * @throws WikiComponentException if the document can't be retrieved
      */
     boolean hasProgrammingRights(DocumentReference reference) throws WikiComponentException;
+
+    /**
+     * Get the XWikiDocument corresponding to the given Document Reference.
+     *
+     * @param reference the document reference
+     * @return the found XWikiDocument
+     * @throws WikiComponentException if an error occurs
+     */
+    XWikiDocument getDocument(DocumentReference reference) throws WikiComponentException;
```

Comment:
The purpose of WikiComponentBridge is to hide oldcore API, this method is doing quite the opposite of that :)